src/transformers/tokenization_utils_base.py

```diff
@@ -1742,16 +1744,34 @@ def _from_pretrained(
         # Prepare tokenizer initialization kwargs
         # Did we saved some inputs and kwargs to reload ?
         tokenizer_config_file = resolved_vocab_files.pop("tokenizer_config_file", None)
+        config_tokenizer_class = None
         if tokenizer_config_file is not None:
             with open(tokenizer_config_file, encoding="utf-8") as tokenizer_config_handle:
                 init_kwargs = json.load(tokenizer_config_handle)
+            config_tokenizer_class = init_kwargs.get("tokenizer_class")
             init_kwargs.pop("tokenizer_class", None)
             saved_init_inputs = init_kwargs.pop("init_inputs", ())
             if not init_inputs:
                 init_inputs = saved_init_inputs
         else:
             init_kwargs = init_configuration
 
+        if tokenizer_config_file is None or config_tokenizer_class is None:
+            config_file = resolved_vocab_files.pop("config_file", None)
+            if config_file is not None:
+                with open(config_file, encoding="utf-8") as config_handle:
+                    config_dict = json.load(config_handle)
+                config_tokenizer_class = config_dict.get("tokenizer_class")
```

Comment:
We should rely on `AutoConfig.from_pretrained` for this blob (inside a try block).

Reply:
Thank you for your review. It seems better than my code, and I avoided a circular import by importing `AutoConfig` inside `_from_pretrained` not at the top level (as https://github.com/huggingface/transformers/pull/12619#discussion_r668631856 showed).